crates/ide-assists/src/handlers/convert_two_arm_bool_match_to_matches_macro.rs

```diff
@@ -0,0 +1,258 @@
+use syntax::ast::{self, AstNode};
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+// Assist: convert_two_arm_bool_match_to_matches_macro
+//
+// Convert 2-arm match that evaluates to a boolean into the equivalent matches! invocation.
+//
+// ```
+// fn main() {
+//     match scrutinee$0 {
+//         Some(val) if val.cond() => true,
+//         _ => false,
+//     }
+// }
+// ```
+// ->
+// ```
+// fn main() {
+//     matches!(scrutinee, Some(val) if val.cond())
+// }
+// ```
+pub(crate) fn convert_two_arm_bool_match_to_matches_macro(
+    acc: &mut Assists,
+    ctx: &AssistContext<'_>,
+) -> Option<()> {
+    let match_expr = ctx.find_node_at_offset::<ast::MatchExpr>()?;
+    let match_arm_list = match_expr.match_arm_list()?;
+    if match_arm_list.arms().count() != 2 {
+        cov_mark::hit!(non_two_arm_match);
+        return None;
+    }
+
+    let first_arm = match_arm_list.arms().next()?;
```

Comment:
```suggestion
    let mut arms = match_arm_list.arms();
    let first_arm = arms.next()?;
    let second_arm = arms.next()?;
    if arms.next().is_some() {
        cov_mark::hit!(non_two_arm_match);
        return None;
    }
```

Also apologies, I made a mistake when I said
> That makes this code here even simple, we don't need to bother looking at the second arm at all.

what i meant to say as we do not need to inspect the other arms pattern, we do need to check if the body of the other arm is a `false` literal though, since this could be an arbitrary expression which would make changing this to `matches!` change behavior